integration_tests/compose_deployment/splunk_test_utils_compose.py

```diff
@@ -0,0 +1,182 @@
+    with open("../inventory-tests.csv", "w") as fp:
+        fp.write(result)
+
+
+def update_profiles(profiles):
+    yaml = ruamel.yaml.YAML()
+    with open("../scheduler-config.yaml") as f_tmp:
+        scheduler_config = yaml.load(f_tmp)
+    scheduler_config["profiles"] = profiles
+    with open("../scheduler-config.yaml", "w") as file:
+        yaml.dump(scheduler_config, file)
+
+
+def update_groups(groups):
+    yaml = ruamel.yaml.YAML()
+    with open("../scheduler-config.yaml") as f_tmp:
+        scheduler_config = yaml.load(f_tmp)
+    scheduler_config["groups"] = groups
+    with open("../scheduler-config.yaml", "w") as file:
+        yaml.dump(scheduler_config, file)
+
+
+def update_traps_secrets(secrets):
+    yaml = ruamel.yaml.YAML()
+    with open("../traps-config.yaml") as f_tmp:
+        traps_config = yaml.load(f_tmp)
+    traps_config["usernameSecrets"] = secrets
+    with open("../traps-config.yaml", "w") as file:
+        yaml.dump(traps_config, file)
+
+
+def upgrade_docker_compose():
+    os.system("cd .. && sudo docker compose $(find docker* | sed -e 's/^/-f /') up -d")
+
+
+def create_v3_secrets(
+    secret_name="secretv4",
+    user_name="snmp-poller",
+    auth_key="PASSWORD1",
+    priv_key="PASSWORD1",
+    auth_protocol="SHA",
+    priv_protocol="AES",
+):
+    os.system(
+        f'python3 $(realpath "../manage_secrets.py") --path_to_compose $(pwd)/.. \
+    --secret_name {secret_name} \
+    --userName {user_name} \
+    --privProtocol {priv_protocol} \
+    --privKey {priv_key} \
+    --authProtocol {auth_protocol} \
+    --authKey {auth_key} \
+    --contextEngineId 8000000903000A397056B8AC'
+    )
+
+
+def wait_for_containers_initialization():
+    script_body = """ 
+    while true; do
+        CONTAINERS_SC4SNMP=$(sudo docker ps | grep "sc4snmp\\|worker-poller\\|worker-sender\\|worker-trap" | grep -v "Name" | wc -l)
+        if [ "$CONTAINERS_SC4SNMP" -gt 0 ]; then
+          CONTAINERS_UP=$(sudo docker ps | grep "sc4snmp\\|worker-poller\\|worker-sender\\|worker-trap" | grep "Up" | wc -l)
+          CONTAINERS_EXITED=$(sudo docker ps | grep "sc4snmp\\|worker-poller\\|worker-sender\\|worker-trap" | grep "Exited" | wc -l)
+          CONTAINERS_TOTAL=$CONTAINERS_SC4SNMP
+    
+          if [ "$CONTAINERS_UP" -eq "$CONTAINERS_TOTAL" ] || \
+             { [ "$CONTAINERS_EXITED" -eq 1 ] && [ "$((CONTAINERS_UP + CONTAINERS_EXITED))" -eq "$CONTAINERS_TOTAL" ]; }; then
+            echo $(green "All 'sc4snmp' containers are ready.")
+            break
+          fi
+    
+          echo $(yellow "Waiting for all 'sc4snmp' containers to be ready...")
+        else
+          echo $(yellow "No 'sc4snmp' containers found. Waiting for them to appear...")
+        fi
+        sleep 1
+    done 
+    """
+    with open("check_for_containers.sh", "w") as fp:
+        fp.write(script_body)
+    os.system("chmod a+x check_for_containers.sh && ./check_for_containers.sh")
+
+
+# if __name__ == "__main__":
```

Comment:
is this part for local test?

Reply:
No, this is run in the pipeline. We just save this data to file and run this file. I copied this solution from the similar function used in tests for microk8s deployment https://github.com/splunk/splunk-connect-for-snmp/blob/main/integration_tests/splunk_test_utils.py#L180